components/aws/aws.app.js

```diff
@@ -1,27 +1,202 @@
+      default: "us-east-1",
+    },
+  },
   methods: {
+    /**
+     * This function configures and returns a new reference to the AWS SDK. The
+     * configuration involves setting the region based on the provided argument,
+     * as well as setting up the credentials previously provided by the user
+     * when setting up their AWS account.
+     *
+     * @param {string} region - The AWS region to which the AWS SDK will
+     * connect. This string should be an acceptable value by the AWS SDK, which
+     * you can find in the ${@linkcode ./regions.js regions.js} file.
+     * @returns A new configured instance of the AWS SDK
+     */
     sdk(region) {
-      process.env.AWS_ACCESS_KEY_ID = this.$auth.accessKeyId;
-      process.env.AWS_SECRET_ACCESS_KEY = this.$auth.secretAccessKey;
       const AWS = require("aws-sdk");
-      AWS.config.update({ region });
+      const {
+        accessKeyId,
+        secretAccessKey,
+      } = this.$auth;
+      AWS.config.update({
+        credentials: {
+          accessKeyId,
+          secretAccessKey,
+        },
+        region,
+      });
       return AWS;
     },
+    _getIamClient(region) {
+      const AWS = this.sdk(region);
+      return new AWS.IAM();
+    },
+    _getAssumeRolePolicyForService(service) {
+      return {
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Effect: "Allow",
+            Principal: {
+              Service: service,
+            },
+            Action: "sts:AssumeRole",
+          },
+        ],
+      };
+    },
+    /**
+     * This method creates an IAM role for an AWS service. The role will be
+     * created in the specified region, and will grant `sts:AssumeRole`
+     * permissions to the referred service.
+     *
+     * @param {string} region - The AWS region to which the AWS SDK will
+     * connect. This string should be an acceptable value by the AWS SDK, which
+     * you can find in the ${@linkcode ./regions.js regions.js} file.
+     * @param {string}  service - The service that will be granted
+     * `sts:AssumeRole` permissions (e.g. `states.amazonaws.com` for Step
+     * Functions). You can see [this
+     * link]{@link https://gist.github.com/shortjared/4c1e3fe52bdfa47522cfe5b41e5d6f22}
+     * for a community-driven list (the official AWS docs still don't have such
+     * a list).
+     * @param {string} [roleDescription] - A text describing the purpose of the
+     * role
+     * @returns {Promise<object>} An object containing the new role's name and
+     * ARN (`roleName` and `roleArn` respectively)
+     */
+    async createServiceRole(region, service, roleDescription) {
+      const assumeRolePolicyDocument = JSON.stringify(
+        this._getAssumeRolePolicyForService(service),
+      );
+      const params = {
+        AssumeRolePolicyDocument: assumeRolePolicyDocument,
+        RoleName: generateRandomUniqueName(),
+        Description: roleDescription,
+        Path: "/pipedream/",
```

Comment:
Does the `path` appear with the name of the role in the IAM console? Should we prefix the `name` with "pipedream-" or some string to clarify that the role was created by PD?

Reply:
It doesn't appear in the console unless you go into the role itself.
The output of the `generateRandomUniqueName` function is already prefixed with `pd-`. Should I change that to `pipedream-` instead?

Reply:
I see, nah `pd-` is totally fine, thanks